runtime/tests/interpreter/fixedpoint_test.go

```diff
@@ -592,3 +594,115 @@ func TestInterpretFixedPointMinMax(t *testing.T) {
 		})
 	}
 }
+
+func TestStringFixedpointConversion(t *testing.T) {
+	type testcase struct {
+		decimal    *big.Int
+		fractional *big.Int
+	}
+
+	type testsuite struct {
+		name         string
+		toFixedValue func(isNegative bool, decimal, fractional *big.Int, scale uint) (interpreter.Value, error)
+		intBounds    []*big.Int
+		fracBounds   []*big.Int
+	}
+
+	suites := []testsuite{
+		{
+			"Fix64",
+			func(isNeg bool, decimal, fractional *big.Int, scale uint) (interpreter.Value, error) {
+				fixedVal, err := fixedpoint.NewFix64(isNeg, decimal, fractional, scale)
+				if err != nil {
+					return nil, err
+				}
+				return interpreter.NewUnmeteredFix64Value(fixedVal.Int64()), nil
+			},
+			[]*big.Int{sema.Fix64TypeMinIntBig, sema.Fix64TypeMaxIntBig},
+			[]*big.Int{sema.UFix64TypeMinFractionalBig, sema.Fix64TypeMaxFractionalBig},
+		},
+		{
+			"UFix64",
+			func(_ bool, decimal, fractional *big.Int, scale uint) (interpreter.Value, error) {
+				fixedVal, err := fixedpoint.NewUFix64(decimal, fractional, scale)
+				if err != nil {
+					return nil, err
+				}
+				return interpreter.NewUnmeteredUFix64Value(fixedVal.Uint64()), nil
+			},
+			[]*big.Int{sema.Fix64TypeMinIntBig, sema.Fix64TypeMaxIntBig},
+			[]*big.Int{sema.UFix64TypeMinFractionalBig, sema.UFix64TypeMaxFractionalBig},
+		},
+	}
+
+	genCases := func(intComponents, fracComponents []*big.Int) []testcase {
+		testcases := []testcase{
+			{big.NewInt(10), big.NewInt(11)},
+			{big.NewInt(420), big.NewInt(840)},
+			{big.NewInt(123), big.NewInt(45)},
+		}
+
+		for _, intPart := range intComponents {
+			for _, fracPart := range fracComponents {
+				clonedInt := new(big.Int).Set(intPart)
+				clonedFrac := new(big.Int).Set(fracPart)
+				testcases = append(testcases, testcase{clonedInt, clonedFrac})
+			}
+		}
+		return testcases
+	}
+
+	bigZero := big.NewInt(0)
+
+	getMagnitude := func(n *big.Int) uint {
+		var m uint
+
+		cloned := new(big.Int).Set(n)
+		bigTen := big.NewInt(10)
+
+		for m = 0; cloned.Cmp(bigZero) != 0; m++ {
+			cloned.Div(cloned, bigTen)
+		}
+
+		return m
+	}
+
+	for _, testsuite := range suites {
+		t.Run(testsuite.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
Subtle bug here: The `for`-`range` loop *rebinds* `testsuite` to each suite, so by the time the sub-test closure is run, it might already have a different value assigned to it. 

You can fix this by running the tests sequentially, by removing the t.Parallel call; or by rebinding `testsuite` – the common way in the code base is to refactor the loop-body into a local helper function:

```go
test := func (testsuite testsuite) {
	t.Run(testsuite.name, func(t *testing.T) {
		t.Parallel()
		// ...
	})
}

for _, testsuite := range suites {
	test(testsuite)
}
```

Reply:
what in the actual python